### PR TITLE
Feature: remove item

### PR DESCRIPTION
**Pull request by anttibull**

### CODE REVIEW

File: src/tests/unit/Buttons.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  it("should get confirm button, when block with buttons shown", () => {
+    const confirm = buttonsDriver.given
+      .showButtons({ show: true })
+      .when.created()
+      .then.confirmBtnExist();
+    expect(confirm).toBeTruthy();
+  });
+
+  it("should get cancel button, when block with buttons shown", () => {
```
Comment (maxvektor) [inline src/tests/unit/Buttons.spec.js:33]:
```suggestion
  it("should show cancel button...", () => {
```

File: src/tests/unit/Buttons.spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const cancel = buttonsDriver.given
+      .showButtons({ show: true })
+      .when.created()
+      .then.cancelBtnExist();
+    expect(cancel).toBeTruthy();
+  });
+
+  it("should get the icon on confirm button, when block with buttons shown", () => {
```
Comment (maxvektor) [inline src/tests/unit/Buttons.spec.js:41]: change this and one below to click tests

File: src/components/Buttons/Buttons.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          className={styles.confirmationWrapper}
+          data-testid={testIds.confirmationWrapper}
+        >
+          <button
+            className={styles.btnConfirm}
+            data-testid={testIds.btnConfirm}
+            onClick={(event) => handleClick(event)}
+            value="confirm"
+          >
+            &nbsp;
```
Comment (maxvektor) [inline src/components/Buttons/Buttons.jsx:38]: Why?) use css pls

File: src/components/Buttons/Buttons.module.css [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    display: flex;
+    align-items: center;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: space-between;
+
+    width: 70px;
+}
+
+.btnConfirm,
```
Comment (maxvektor) [inline src/components/Buttons/Buttons.module.css:29]: you can use one class for those buttons

File: src/components/Table/Table.tsx [hunk 5]
```diff
@@ -2,19 +2,22 @@ import React from "react";
...
+  amount: number;
+  id: string;
+};
 
- type IProps = {
-    operationList: IExpance[]
-}
+type IProps = {
+  operationList: IExpance[];
+  removeItem: Function;
```
Comment (maxvektor) [inline src/components/Table/Table.tsx:17]:
```suggestion
  removeItem: (id: string) => void;
```
`Function` is a bad practice

File: src/components/Buttons/Buttons.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+
+import { testIds } from "./testIds.js";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";
+
+import styles from "./Buttons.module.css";
+
+export const Buttons = ({ onClick, show = false }) => {
```
Comment (maxvektor) [inline src/components/Buttons/Buttons.jsx:10]:
```suggestion
export const Buttons = ({ onClick, confirm=false, text}) => {
```

File: src/components/Buttons/Buttons.unit.driver.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      },
+    },
+    when: {
+      created: () => {
+        _wrapper = render(<Buttons {..._props} />);
+        return driver;
+      },
+    },
+    then: {
+      confirmationWrapperExist: () =>
```
Comment (maxvektor) [inline src/components/Buttons/Buttons.unit.driver.js:39]:
```suggestion
      exist: () => 
```

File: src/components/Table/Table.tsx [hunk 8]
```diff
@@ -26,6 +29,10 @@ export const Table = ({ operationList }: IProps) => {
             <td className={table.amount} data-testid={testIds.amount}>
               {item.amount}
             </td>
+            <div> </div>
```
Comment (maxvektor) [inline src/components/Table/Table.tsx:32]: remove

File: src/components/Table/Table.tsx [hunk 9]
```diff
@@ -2,22 +2,28 @@ import React from "react";
...
-  removeItem: Function;
+  removeItem: (id: string) => void;
+  withConfirm?: boolean;
 };
 
-export const Table = ({ operationList, removeItem }: IProps) => {
+export const Table = ({
+  operationList,
+  removeItem,
+  withConfirm = true,
```
Comment (anttibull) [inline src/components/Table/Table.tsx:25]: this is probably bad practice, but did not understand how to do it better

Comment (maxvektor) [reply src/components/Table/Table.tsx:25]:
```suggestion

```

File: src/components/Table/Table.tsx [hunk 10]
```diff
@@ -26,6 +35,13 @@ export const Table = ({ operationList }: IProps) => {
             <td className={table.amount} data-testid={testIds.amount}>
               {item.amount}
             </td>
+            <td className={table.removeItem}>
+              <Button
+                onClick={() => removeItem(item.id)}
+                withConfirm={withConfirm}
```
Comment (maxvektor) [inline src/components/Table/Table.tsx:41]:
```suggestion
                withConfirm={true}
```

File: src/tests/unit/Buttons.spec.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+  it("should be hidden block with the confirmation and cancel buttons", () => {
+    const confirmationWrapper = buttonsDriver.given
+      .props()
+      .when.created()
+      .then.confirmationWrapperExist();
+    expect(confirmationWrapper).toBeFalsy();
+  });
+
+  it("should get confirm button, when block with buttons shown", () => {
```
[the remark this answers is not in the document]
Comment (anttibull) [reply src/tests/unit/Buttons.spec.js]: we don’t need this test or need to change it?

**Pull request closed without merging**